Classes/Domain/Repository/GlossaryRepository.php

```diff
@@ -470,33 +471,47 @@ private function getGlossary(
      */
     private function getGlossariesInRootByCurrentPage(int $pageId): array
     {
-        $site = GeneralUtility::makeInstance(SiteFinder::class)
-            ->getSiteByPageId($pageId);
-        $rootPage = $site->getRootPageId();
-        $allPages = GeneralUtility::makeInstance(PageTreeRepository::class)
-            ->getTreeList($rootPage, 999);
         $db = GeneralUtility::makeInstance(ConnectionPool::class)
             ->getQueryBuilderForTable('pages');
-        $statement = $db
+
+        $result = $db
             ->select('uid')
             ->from('pages')
             ->where(
-                $db->expr()->in('uid', $allPages),
-                $db->expr()->eq('doktype', $db->createNamedParameter(254, Connection::PARAM_INT)),
+                $db->expr()->eq(
+                    'doktype',
+                    $db->createNamedParameter(
+                        PageRepository::DOKTYPE_SYSFOLDER,
+                        Connection::PARAM_INT
+                    )
+                ),
                 $db->expr()->eq('module', $db->createNamedParameter('glossary'))
-            );
-        $result = $statement->executeQuery()->fetchAllAssociative();
+            )->executeQuery();
 
-        if (!is_array($result)) {
+        if ($result->rowCount() === 0) {
```

Comment:
SELECT->rowCount() should not be used. Either usa a dedicated count query first, or early return is not possible.

But instead of fetchAllAssociative() the core way is using `while($row = $result->fetchAssociative()) {}` instead of a foreach on all retrieved records.